Userland/chmod.cpp

```diff
@@ -3,18 +3,53 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+int str_to_octal_mode(char access_scope, char** access_str, unsigned& mode);
```

Comment:
In Serenity we generally try to use complete words instead of abbreviations in function names and variables.
The main exception is i/j/it/jt for iterators.

Also we can make use of `Optional<T>` for the return value:
```cpp
#include <AK/Optional.h>
...
Optional<mode_t> string_to_mode(char access_scope, char** access_string);
```

Then you can return `{}` in the case of failure, and just a regular `mode_t` in successful cases.
The client then just checks `.has_value()` on the returned value to see if it's valid or not.